test/test-ping-pong.c

```diff
@@ -252,6 +303,75 @@ static void pipe_pinger_new(int vectored_writes) {
 }
 
 
+static void socketpair_pinger_new(int vectored_writes) {
+  pinger_t* pinger;
+  uv_os_sock_t fds[2];
+  uv_tcp_t* ponger;
+
+  pinger = (pinger_t*) malloc(sizeof(*pinger));
+  ASSERT(pinger != NULL);
+  pinger->vectored_writes = vectored_writes;
+  pinger->state = 0;
+  pinger->pongs = 0;
+  pinger->pong = PONG;
+
+  /* Try to make a socketpair and do NUM_PINGS ping-pongs. */
+  (void)uv_default_loop(); /* ensure WSAStartup has been performed */
+  ASSERT(0 == uv_socketpair(SOCK_STREAM, 0, fds, UV_NONBLOCK_PIPE, UV_NONBLOCK_PIPE));
+#ifndef _WIN32
+  /* On Windows, this is actually a UV_TCP, but libuv doesn't detect that. */
+  ASSERT(uv_guess_handle((uv_file) fds[0]) == UV_NAMED_PIPE);
+  ASSERT(uv_guess_handle((uv_file) fds[1]) == UV_NAMED_PIPE);
+#endif
+
+  ASSERT(0 == uv_tcp_init(uv_default_loop(), &pinger->stream.tcp));
+  pinger->stream.pipe.data = pinger;
+  ASSERT(0 == uv_tcp_open(&pinger->stream.tcp, fds[1]));
+
+  ponger = (uv_tcp_t*) malloc(sizeof(uv_tcp_t));
+  ASSERT(ponger != NULL);
+  ponger->data = NULL;
+  ASSERT(0 == uv_tcp_init(uv_default_loop(), ponger));
+  ASSERT(0 == uv_tcp_open(ponger, fds[0]));
+
+  pinger_write_ping(pinger);
+
+  uv_read_start((uv_stream_t*) &pinger->stream.tcp, alloc_cb, pinger_read_cb);
+  uv_read_start((uv_stream_t*) ponger, alloc_cb, ponger_read_cb);
```

Comment:
Can you `ASSERT_EQ(0, uv_read_start(...));` here?

Similar suggestions apply to the function below.